presto-spi/src/main/java/com/facebook/presto/spi/function/Signature.java

```diff
@@ -168,6 +168,17 @@ public String toString()
                 "(" + String.join(",", argumentTypes.stream().map(TypeSignature::toString).collect(toList())) + "):" + returnType;
     }
 
+    /*
```

Comment:
I'm not sure if this is really a 'canonical' form. At any rate, since this only finds used in the ScalarFormAnnotationsParser (and other consumers) can we instead move it there as a static method ?

Reply:
For example a scalar string function signature such as `varchar(y) fun1(varchar(x) , varchar(z))` since the values x, y, z depend on invocation of this scalar function (e.g. `varchar(20) fun1(varchar(10) , varchar(10))` ). Now to convert it into a form that can be equated, it is be transformed to `varchar fun1(varchar , varchar)`. This is the reason for calling it a canonical form (since the original function signature needs to be compared regardless of how it was invoked). Feel free to correct me on this and I am open to a better naming.